Feature/add pupil page

PReichetanz opened the pull request:
Bringing together all components and add styling:

- Rename MainButton to Button for better handling
- Make Form display dependent on state via buttons
- Add container with grid layout
- Add header
- Add backButton to Form (closes #12, but was not part of the user story) 

<div>
<img width=200 src="https://user-images.githubusercontent.com/33567019/143456453-f6fa8e01-dc07-48ab-a132-b32eb167e77e.png">
<img width=200 src="https://user-images.githubusercontent.com/33567019/143456470-1680e91d-8e42-4819-8a02-921d8f287d29.png">
</div>

[View on heroku](https://capstone-rev-feature-ad-b834ot.herokuapp.com/)
[View Storybook](https://capstone-rev-feature-ad-b834ot.herokuapp.com/storybook/?path=/story/component-button--regular)

closes #2

PReichetanz committed:
Add return to main view to form

PReichetanz committed:
Rename MainButton to Button

- Rename all imports
- Adjust story (also with general onClick function)

PReichetanz committed:
Add absolute positioning to Form

PReichetanz committed:
Update App.tsx

- Create displayState
- Add Container with grid for layout
- Add header and styling
- Make form rendering conditional from displayState via AddButton
- Add styling

PReichetanz committed:
Update Form story with onCancel

lajanzen commented on src/app/App.tsx:
```diff
@@ -1,15 +1,17 @@
+export default function App(): JSX.Element {
   const [pupils, setPupils] = useState<Pupil[]>([]);
-  console.log(pupils);
+  const [displayState, setDisplayState] = useState('');
```
It seems like you only have 2 states here, so I would rather work with true/false: 

```suggestion
  const [isFormShown, setIsFormShown] = useState(false);
```

lajanzen commented on src/app/App.tsx:
```diff
@@ -36,6 +38,7 @@ function App(): JSX.Element {
           evaluations: [pupil.evaluation],
         },
       ]);
+      setDisplayState('');
```
```suggestion
      setIsFormShown(false);
```

lajanzen commented on src/app/App.tsx:
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
+        <h1>Meine Klasse</h1>
+      </Header>
+      <Main>
+        {displayState === 'showAddPupilForm' && (
```
```suggestion
        {isFormShown && (
```

lajanzen commented on src/app/App.tsx:
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
+            evaluationLabel="Worturteil"
+            onSubmit={handleFormSubmit}
+            submitted={false}
+            onCancel={() => setDisplayState('')}
```
```suggestion
            onCancel={() => setIsFormShown(false)}
```

lajanzen commented on src/app/App.tsx:
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
+          />
+        ))}
+      </Main>
+      <AddButton onClick={() => setDisplayState('showAddPupilForm')}>
```
```suggestion
      <AddButton onClick={() => setIsFormShown(true)}>
```

lajanzen commented on src/app/App.tsx:
```diff
@@ -45,22 +48,60 @@ function App(): JSX.Element {
+const Container = styled.div`
+  height: 100vh;
+  display: grid;
+  grid-template-columns: 1fr;
```
Do you need this?
```suggestion
```

PReichetanz replied:
I don't, thanks!

derjubu commented on src/app/components/Button/Button.stories.tsx:
```diff
@@ -0,0 +1,11 @@
+};
+
+export const Regular = (): JSX.Element => (
+  <Button onClick={() => console.log('clicked!')}>Schüler hinzufügen</Button>
```
I think you can remove this onClick function, if it only logs something in the console for control purposes during development.

PReichetanz replied:
I really would like to, but TypeScript needs it to be happy!

lajanzen commented on src/app/components/Form/Form.tsx:
```diff
@@ -32,44 +35,66 @@ export default function Form({
+          placeholder="Lena Beispiel"
+          onChange={(event) => setName(event.target.value)}
+          value={name}
+          submitted={inputError}
```
What does this do exactly?

PReichetanz replied:
It checks whether an input is missing after submitting. I renamed it to "missingInput", so it should be more descriptive.

PReichetanz committed:
Add changes according to review

- Rename state for display form to "isFormShown"
- Rename prop of Form "submitted" to "missingInput"
- Remove unnecessary grid-template-column with only 1fr
